pandas/core/groupby/generic.py

```diff
@@ -1200,158 +1200,147 @@ def first_not_none(values):
             # GH9684. If all values are None, then this will throw an error.
             # We'd prefer it return an empty dataframe.
             return DataFrame()
-        elif isinstance(v, DataFrame):
-            return self._concat_objects(keys, values, not_indexed_same=not_indexed_same)
-        elif self.grouper.groupings is not None:
```

Comment:
Little hard to tell from the diff - is removing this branch the only change?

Reply:
I also changed the `elif`s after the `return` statements to `if` so that I should shift the previous branch. Shall I only remove the last branch if that makes it easier to review?